### PR TITLE
Normalize CODEOWNERS entries to use -approvers

### PR DESCRIPTION
**Pull request by chalin**

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -20,17 +20,17 @@ content/en/blog/                         @open-telemetry/docs-approvers @open-te
 content/en/community/end-user/           @open-telemetry/docs-approvers @open-telemetry/end-user-wg
 content/en/docs/collector                @open-telemetry/docs-approvers @open-telemetry/collector-approvers
 content/en/docs/demo                     @open-telemetry/docs-approvers @open-telemetry/demo-approvers
-content/en/docs/instrumentation/cpp/     @open-telemetry/docs-approvers @open-telemetry/cpp-maintainers
+content/en/docs/instrumentation/cpp/     @open-telemetry/docs-approvers @open-telemetry/cpp-approvers
 content/en/docs/instrumentation/erlang/  @open-telemetry/docs-approvers @open-telemetry/erlang-approvers
 content/en/docs/instrumentation/go/      @open-telemetry/docs-approvers @open-telemetry/go-instrumentation-approvers
```
Comment (svrnm) [inline .github/CODEOWNERS:25]:
```suggestion
content/en/docs/instrumentation/go/      @open-telemetry/docs-approvers @open-telemetry/go-approvers @open-telemetry/go-instrumentation-approvers
```

**Comment by svrnm** [comment]
Are we sure that the approvers always contain the maintainers as well?

**Comment by chalin** [comment]
Actually, if it isn't the case, then the team rosters should be fix (IMHO).

**Comment by svrnm** [comment]
if you checked a few then I hope we are safe here:-)

**Comment by chalin** [comment]
Yes, I cross-checked the following (so all those affected by this change), and the approvers include the maintainers via subteam inclusion:

- cpp
- java (main and instrumentation)
- php
- python
- rust
- operator
- demo

**Pull request merged**